protocol/rpcprovider/provider_listener_test.go

```diff
@@ -0,0 +1,39 @@
+package rpcprovider
+
+import (
+	"context"
+	"testing"
+
+	"github.com/lavanet/lava/v3/protocol/lavasession"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/health/grpc_health_v1"
+)
+
+func TestHealthCheck(t *testing.T) {
+	address := "0.0.0.0:2220"
+
+	networkAddressData := lavasession.NetworkAddressData{
+		Address:    address,
+		DisableTLS: true,
+	}
+
+	providerListener := NewProviderListener(context.Background(), networkAddressData, "/lava/health")
+
+	defer providerListener.Shutdown(context.Background())
+
+	conn, err := grpc.NewClient(address, grpc.WithTransportCredentials(insecure.NewCredentials()))
+	if err != nil {
+		t.Fatalf("GRPC new client failed: %v", err)
+	}
```

Comment:
Use the require lib (don't forget to import it)
```suggestion
    require.NoError(t, err)
```